src/pages/edge/adobe-journey-optimizer/in-app-message/tutorials/tabs/javascript-from-native.md

```diff
@@ -1,4 +1,40 @@
-<Variant platform="android" task="obtain" repeat="3"/>
+<Variant platform="android3x" task="obtain" repeat="5"/>
+
+In the `onShow` function of the `PresentationDelegate`, get a reference to the `InAppEventHandler` which can be used for Javascript interactions.
+
+#### Kotlin
+
+```kotlin
+var eventHandler: InAppMessageEventHandler? = null
+var currentMessagePresentable: Presentable<InAppMessage>? = null
+
+override fun onShow(presentable: Presentable<*>) {
+  if (presentable.getPresentation() !is InAppMessage) {
+    return
+  }
+  currentMessagePresentable = presentable as Presentable<InAppMessage>
+  eventHandler = currentMessagePresentable?.getPresentation()?.eventHandler
+
+}
+```
+
+#### Java
+
+```java
+InAppMessageEventHandler eventHandler = null;
+Presentable<InAppMessage> currentMessagePresentable = null;
+
+@Override
+public void onShow(Presentable<?> presentable) {
+    if (presentable.getPresentation() !is InAppMessage) {
```

Comment:
This should be changed to `instanceOf`. Same comment for`evaluateJavascript` java example